packages/lab/src/stepper-input/internal/useSpinner.ts

```diff
@@ -1,33 +1,49 @@
-import { useEffect, useState } from "react";
-
+import { useEffect, useState, SyntheticEvent, useCallback } from "react";
 import { useInterval } from "./useInterval";
 
-const INTERVAL_DELAY = 300;
+const INITIAL_DELAY = 500;
+const INTERVAL_DELAY = 100;
 
-function useSpinner(activationFn: () => void, isAtLimit: boolean) {
+function useSpinner(
+  activationFn: (event?: SyntheticEvent) => void,
+  isAtLimit: boolean
+) {
   const [buttonDown, setButtonDown] = useState(false);
+  const [delay, setDelay] = useState(INITIAL_DELAY);
 
-  const cancelInterval = () => setButtonDown(false);
+  const cancelInterval = () => {
```

Comment:
should this be a callback?

Reply:
honestly not sure when things should be a callback

Reply:
ESLint says: The 'cancelInterval' function makes the dependencies of useEffect Hook (at line 28) change on every render. To fix this, wrap the definition of 'cancelInterval' in its own useCallback() Hook. (react-hooks/exhaustive-deps)